_analyzers/token-filters/hyphenation-decompounder.md

```diff
@@ -0,0 +1,102 @@
+---
+layout: default
+title: Hyphenation decompounder
+parent: Token filters
+nav_order: 170
+---
+
+# Hyphenation decompounder token filter
+
+The `hyphenation_decompounder` token filter is used to break down compound words into their constituent parts. This filter is particularly useful for languages like German, Dutch, and Swedish, where compound words are common. The filter uses hyphenation patterns (typically defined in .xml files) to identify the possible locations within a compound word where it can be split into components. These components are then checked against a provided dictionary. If there is a match, those components are treated as valid tokens. For more information about hyphenation pattern files, see [FOP XML Hyphenation Patterns](https://offo.sourceforge.net/#FOP+XML+Hyphenation+Patterns).
+
+## Parameters
+
+The `hyphenation_decompounder` token filter can be configured with the following parameters.
+
+Parameter | Required/Optional | Data type | Description
+:--- | :--- | :--- | :--- 
+`hyphenation_patterns_path` | Required | String | The path (relative to the `config` directory or absolute) to the hyphenation patterns file, which contains the language-specific rules for word splitting. The file is typically in XML format. Sample files can be downloaded from [OFFO Sourceforge project](https://sourceforge.net/projects/offo/).
+`word_list` | Required if `word_list_path` is not set | Array of strings | A list of words that are used to validate the components generated by the hyphenation patterns.
+`word_list_path` | Required if `word_list` is not set | String | The path (relative to `config` directory or absolute) to a list of subwords.
+`max_subword_size` | Optional | Integer | Maximum subword length. If the generated subword is longer, it will not be added to the generated tokens. Default is `15`.
+`min_subword_size` | Optional | Integer | Minimum subword length. If the generated subword is shorter, it will not be added to the generated tokens. Default is `2`.
```

Comment:
```suggestion
`min_subword_size` | Optional | Integer | The minimum subword length. If the generated subword is shorter than the specified length, it will not be added to the generated tokens. Default is `2`.
```